webapp/src/ee/billing/component/Plan/PlanMigratingChip.tsx

```diff
@@ -0,0 +1,228 @@
+import { Box, Button, Chip, styled, Typography } from '@mui/material';
+import { T, useTranslate } from '@tolgee/react';
+import { useBillingApiQuery } from 'tg.service/http/useQueryApi';
+import { ArrowRight, Settings01 } from '@untitled-ui/icons-react';
+import { PricePrimary } from 'tg.ee.module/billing/component/Price/PricePrimary';
+import React, { useState } from 'react';
+import { SpinnerProgress } from 'tg.component/SpinnerProgress';
+import { FullWidthTooltip } from 'tg.component/common/FullWidthTooltip';
+import { LINKS } from 'tg.constants/links';
+import { Link } from 'react-router-dom';
+import clsx from 'clsx';
+import { PlanType } from 'tg.ee.module/billing/administration/subscriptionPlans/components/migration/types';
+
+const TooltipText = styled('div')`
+  white-space: nowrap;
+`;
+
+const TooltipTitle = styled('div')`
+  font-weight: bold;
+  font-size: 14px;
+  line-height: 17px;
+`;
+
+const MigrationDetailBox = styled(Box)`
+  &.inactive {
+    opacity: 0.5;
+  }
+`;
+
+export const PlanMigratingChip = ({
```

Comment:
This component is very large and should be split into multiple smaller ones.
Also, I consider providing the plan type an anti-pattern. Instead, I recommend creating separate components for SelfHosted and Cloud and another component props for everything which differs to follow "open-closed principle" better.